metrics/metrics.go

```diff
@@ -73,25 +80,47 @@ type CSIMetricsManager interface {
 	StartMetricsEndpoint(metricsAddress, metricsPath string)
 }
 
+// MetricsManagerOption is used to pass optional configuration to a
+// new metrics manager.
+type MetricsManagerOption func(*csiMetricsManager)
+
+// WithSubsystem overrides the default subsystem name.
+func WithSubsystem(subsystem string) MetricsManagerOption {
+	return func(cmm *csiMetricsManager) {
+		cmm.subsystem = subsystem
+	}
+}
+
+// WithStabilityLevel overrides the default stability level.
+func WithStabilityLevel(stabilityLevel metrics.StabilityLevel) MetricsManagerOption {
```

Comment:
Do we want stability level to be configurable? If someone sets it to stable, what happens if the library decides to change the schema?

Reply:
> Do we want stability level to be configurable? If someone sets it to stable, what happens if the library decides to change the schema?

The user cannot use that modified library and has to fork it if it wants to continue supporting the old values. Clearly overriding "alpha" with "stable" is a bad idea for that reason.

I was more interested in the other direction: a downstream user might enable metrics support as an alpha feature and may want to keep the alpha label even when the library moves to "stable".

Reply:
I think a note is enough. If someone really wants to increase stability, they have been warned and will have to live with the consequences.

Reply:
Added.